libc/src/stdlib/CMakeLists.txt

```diff
@@ -331,7 +331,12 @@ if(NOT LIBC_TARGET_OS_IS_GPU)
     include(${LIBC_SOURCE_DIR}/../compiler-rt/cmake/Modules/AllSupportedArchDefs.cmake)
 
     # scudo distinguishes riscv32 and riscv64, so we need to translate the architecture
-    set(LIBC_TARGET_ARCHITECTURE_FOR_SCUDO ${LIBC_TARGET_ARCHITECTURE})
+    # set(LIBC_TARGET_ARCHITECTURE_FOR_SCUDO ${LIBC_TARGET_ARCHITECTURE})
+    if (LIBC_TARGET_OS_IS_DARWIN AND (LIBC_TARGET_ARCHITECTURE STREQUAL "arm"))
+      set(LIBC_TARGET_ARCHITECTURE_FOR_SCUDO arm64)
```

Comment:
Is this because of https://github.com/llvm/llvm-project/blob/c1b70fa5bfea973d4141e27cf9668e9325609e19/libc/cmake/modules/LLVMLibCArchitectures.cmake#L37-L38? Should that be fixed instead?

Reply:
It is because of that.  Let me try to see if correctly fixing that would be too big for this PR or not.

Reply:
Cleaning that up would be quite invasive and too big for this PR.  I'd leave that for the follow up issue / PR.